external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -110,6 +110,34 @@ private[kafka010] class KafkaOffsetReader(
     kafkaReaderThread.shutdown()
   }
 
+  /**
+   * Commits offsets to the consumer_offsets topic
+   * for the driver consumer.
+  */
+  def commitOffsetsAsync(
+      partitionToOffsets: Map[TopicPartition, Long],
+      s: KafkaCommitsSource)
+    : Unit = {
+    if (_consumer != null) runUninterruptibly {
+     val commitCallback = new OffsetCommitCallback() {
+        override def onComplete(
+            offsets: ju.Map[TopicPartition,
+            OffsetAndMetadata], exception: Exception)
+        : Unit = {
+          if (exception != null) {
+            log.warn("Committing offsets to Kafka failed. This does " +
```

Comment:
Instead of saying what not compromise I would tell what is it used for. See my previous comment in the doc area.